scripts/recovery_sync

```diff
@@ -9,6 +9,11 @@
 
 set -euxo pipefail
 
+function cleanup() {
+	rm -rf "$workdir" || true
+	rm "${target}.tmp" || true
```

Comment:
I don't think you need `|| true` here. Using `-f` ensures that the command doesn't fail if the file/directory doesn't exist. It will fail however if the file exists but it somehow failed to delete it, which could expose some other issues.